starlette/websockets.py

```diff
@@ -66,12 +73,19 @@ async def send(self, message: Message) -> None:
         if self.application_state == WebSocketState.CONNECTING:
             message_type = message["type"]
             if message_type not in {"websocket.accept", "websocket.close"}:
-                raise RuntimeError(
-                    'Expected ASGI message "websocket.accept" or '
-                    f'"websocket.close", but got {message_type!r}'
-                )
+                if not (
+                    self._have_response_extension()
+                    and message_type == "websocket.http.response.start"
+                ):
+                    raise RuntimeError(
+                        'Expected ASGI message "websocket.accept"'
+                        ' or "websocket.close", '
+                        f"but got {message_type!r}"
+                    )
```

Comment:
Do we have a case where we don't have the extension, and we send a `"websocket.http.response.start"`?

The only place we send a `websocket.http.response.start` is on `send_denial_response`, isn't it?

Reply:
Yes, that is true, but `send()` is already doing a lot of state checking.  I guess we can remove the check from here.